phpstan-baseline.neon

```diff
@@ -0,0 +1,31 @@
+parameters:
+	ignoreErrors:
+		-
+			message: "#^Call to static method create\\(\\) on an unknown class Symfony\\\\Component\\\\HttpClient\\\\HttpClient\\.$#"
+			count: 1
+			path: src/ClientFactory/SymfonyFactory.php
+
+		-
+			message: "#^Class Http\\\\Client\\\\Plugin\\\\Vcr\\\\NamingStrategy\\\\NamingStrategyInterface not found\\.$#"
+			count: 1
+			path: src/DependencyInjection/Configuration.php
```

Comment:
we could `composer require` the vcr plugin in a first step of the phpstan job, to have it validate correctly. the php tests would discover if we messed up with validation.

some nice to haves would be to have the configuration class error when configuring without vcr without having the plugin installed, and https://github.com/php-http/HttplugBundle/blob/b7ad2c1951edec36ef0bd9897e309b0038c59a54/src/DependencyInjection/HttplugExtension.php should throw an exception at the beginning of the method if vcr is not available.

but probably this still not enough to make phpstan happy, so composer require step for this could be added. that means that phpstan locally would complain about those though :thinking: 

we can also merge the current state and continue in separate pull requests.